chore: document minimal security context settings

jawnsy opened the pull request:
Add documentation that describes how to use opentelemetry-operator in restrictive clusters, such as enabling runAsRoot, dropping capabilities, and configuring seccomp confinement.

Closes: #1264

jawnsy commented on README.md:
```diff
@@ -306,6 +306,54 @@ You can configure the OpenTelemetry SDK for applications which can't currently b
+
+You can optionally configure `runAsUser` and set it to `10001`, as this is the `USER` defined in the [opentelemetry-collector Dockerfile](https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/distributions/otelcol/Dockerfile). In OpenShift, however, configuring this explicitly will conflict with the default `restricted` Security Context Constraint, which runs pods with a pr ...
+
+For a full list of settings, consult the type definition in [opentelemetrycollector_types.go](./apis/v1alpha1/opentelemetrycollector_types.go) or [API docs](./docs/api.md).
```
This line might not be necessary, since the API docs are mentioned at the top. Happy to remove it according to your preference!

pavolloffay replied:
yes, we can remove this line

pavolloffay reviewed (commented):
Thanks for the PR ❤️

frzifus reviewed (commented):
May i ask, whats the current state here? :)

jawnsy commented:
Ah, apologies, I need to clear up the CLA issue before I can proceed

jawnsy commented:
Sorry for the long delay, and thanks for your patience. I've rebased and addressed comments (removed the reference to the API file), and this pull request is now ready for another review.

R011y commented:
Any update on this?

jawnsy commented:
@pavolloffay @frzifus Hey, I've rebased and this should be ready for review when you have a moment!

frzifus commented:
Awesome thanks @jawnsy

R011y commented:
Thanks guys. Greatly appreciated.

R011y commented:
@jawnsy Still need one more for this to get merged <3

R011y commented:
> > Still need one more for this to get merged <3
> 
> @R011y Sorry, can you please clarify -- what are the next steps here? Do I need to make more changes, or are we waiting for another review?

Just needs another review. No changes requested.

R011y commented:
Just for awareness in case others come across this issue: 
Otel operator CAN be run with restrictive security context without issue. It functions correctly with both allowPrivilegeEscalation: false and readOnlyRootFilesystem: true. The documentation update in this PR will just cement the existing ability to run Otel operator with restrictive security context. That said, it's always best to test thoroughly in your environment as there may be conditions or variability that present issues not identified here.

R011y commented:
Add me as a reviewer if you want @jawnsy, if that's possible.

jaronoff97 commented:
@R011y mind reviewing this so i can merge this finally?

atoulme commented:
Closing as inactive. Please comment if you would like to reopen this pull request.

jawnsy commented:
It's unfortunate that this was closed, it seemed ready to merge and only needed reviews?

I'm happy to rebase this if you'll consider reviewing and merging
